### ISSUE TITLE
Download segments of Youtube Dash or HLS format in parallel

### ISSUE THREAD

**Issue by kietheros** [ytdl-org/youtube-dl#26380]
## Checklist

- [x ] I'm asking a question
- [ x] I've looked through the README and FAQ for similar questions
- [ ] I've searched the bugtracker for similar questions including closed ones

## Question

youtube-dl downloads DASH or m3u8 format very slow. Download speed is about 500 kbps, and download speed of other MP4 formats are about 20 Mbps on my computer. Can youtube-dl download segments in parallel. If youtube-dl doesn't support this feature, which method should I use to implement this feature:

- Write a new downloader
- Only use youtube-dl to get list of format info. Use nodejs + ffmpeg to download segments and merge into mp4 video.
